### PR TITLE
Handle list arguments with nargs instead of list type when overriding…

### LINKED ISSUE #4798: valhalla_config_build does not handle list arguments correctly
This issue was previously noted a few years ago, but it appears not to be fixed. See https://github.com/valhalla/valhalla/pull/3225

Essentially, when you build a config file with valhalla_build_config.py, and you pass in arguments to a path with a Python type of `list`, ArgumentParser will apply split() on the argument. That means the argument is streated as a string and splits it into characters, which is not the desired behavior.

Today, the only `list` type argument is `statsd-tags`. 
Example behavior:
```
% ./scripts/valhalla_build_config --statsd-tags list_item_ list_item_2
...
"statsd": {
    "port": 8125,
    "prefix": "valhalla",
    "tags": [
      "[",
      "l",
      "i",
      "s",
      "t",
      "_",
      "i",
      "t",
      "e",
      "m",
      "_",
      ",",
      "l",
      "i",
      "s",
      "t",
      "_",
      "i",
      "t",
      "e",
      "m",
      "_",
      "2",
      "]"
    ]
  },
...
```

The desired output is this:
```
% ./scripts/valhalla_build_config --statsd-tags ["list_item_1", "list_item_2"]
...
"statsd": {
    "port": 8125,
    "prefix": "valhalla",
    "tags": [
      "list_item_1",
      "list_item_2"
    ]
  },
...
```

### PR DESCRIPTION
**Pull request by asmyth01**
# Issue

fixes #4798 

## Tasklist

 - [x] Add tests
 - [x] Add #fixes with the issue number that this PR addresses
 - [x] Update the docs with any new request parameters or changes to behavior described
 - [x] Update the [changelog](CHANGELOG.md)
 - [x] If you made changes to the lua files, update the [taginfo](taginfo.json) too.

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
    * REMOVED: needs_ci_run script [#4423](https://github.com/valhalla/valhalla/pull/4423)
    * REMOVED: unused vehicle types in AutoCost and segway; renamed kTruck to "truck" instead of "tractor_trailer" [#4430](https://github.com/valhalla/valhalla/pull/4430)
    * REMOVED: ./bench and related files/code [#4560](https://github.com/valhalla/valhalla/pull/4560)
+   * FIXED: Handle list type arguments correctly when overriding config with valhalla_build_config [#4799](https://github.com/valhalla/valhalla/pull/4799)
```
Comment (kevinkreiser) [inline CHANGELOG.md:6]: please move this below the bug fix heading

File: scripts/valhalla_build_config [hunk 2]
```diff
@@ -596,7 +596,10 @@ def add_leaf_args(
             t = type(tree)
 
         arg = '--' + path.replace('_', '-').replace('\0', '-')
-        parser_.add_argument(arg, type=t, help=h, default=tree)
+        if t == list:
+            parser_.add_argument(arg, nargs="+", help=h, default=tree)
+        else:
+            parser_.add_argument(arg, type=t, help=h, default=tree)
```
Comment (kevinkreiser) [inline scripts/valhalla_build_config:602]: nit: a ternary would be nicer imho:
```suggestion
        parser_.add_argument(arg, nargs="+" if t == list else t, help=h, default=tree)
```

**Comment by kevinkreiser** [comment]
thank you for this cleanup!

File: scripts/valhalla_build_config (hunk 2 above)
Comment (asmyth01) [reply scripts/valhalla_build_config:602]: @kevinkreiser That would be more concise, but I'm afraid this isn't going to work in this case. Your suggestion would replace the value of `nargs` with the result of the ternary operator. What I really want to do here is use the `nargs` parameter only if `t==list`, else set the paramter `type=t`.

Comment (kevinkreiser) [reply scripts/valhalla_build_config:602]: oh sorry i didnt look closely :smile: ignore the suggestion haha

**Review by kevinkreiser — APPROVED** [review]

**Pull request merged**